File: src/person_service/src/person_server.cpp
#include "ros/ros.h"
#include "person_service/s_c_person.h"//头文件名字与数据类型名字一样

//service回调函数，输入参数req,输出参数res
bool personCallback(person_service::s_c_person::Request &req,person_service::s_c_person::Response &res)
 {
    //显示请求数据
    ROS_INFO("Person:name:%s age:%d sex:%d",req.name.c_str(),req.age,req.sex);

    //设置反馈数据
    res.result="OK";

    return true;
 }
int main(int argc,char**argv)
 {
     //ros节点初始化
     ros::init(argc,argv,"person_server");

     //创建节点句柄
     ros::NodeHandle n;

     //创建一个名为/show_person的server,注册回调函数personCallback
     ros::ServiceServer person_server=n.advertiseService("/show_person",personCallback);

     ROS_INFO("Ready to show person information.");
     //循环等待回调函数
	 ros::spin();

     return 0;
 }

File: src/person_service/scripts/person_server.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# 该例程将请求/show_person服务，服务数据类型person_service::s_c_person

import rospy
from person_service.srv import s_c_person, s_c_personResponse

def personCallback(req):
	# 显示请求数据
    rospy.loginfo("Person: name:%s  age:%d  sex:%d", req.name, req.age, req.sex)

	# 反馈数据
    return s_c_personResponse("OK")

def person_server():
	# ROS节点初始化
    rospy.init_node('person_server')

	# 创建一个名为/show_person的server，注册回调函数personCallback
    s = rospy.Service('/show_person', s_c_person, personCallback)

    print "Ready to show person informtion."

    # 循环等待回调函数
    rospy.spin()

if __name__ == "__main__":
    person_server()


